src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { getUser } from '../../api';
+import { useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
+// import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [user, setUser] = useState<User | null>(null);
 
-      <Loader />
+  const dispatch = useDispatch();
+  const todo = useAppSelector(state => state.currentTodo);
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  useEffect(() => {
+    setIsLoading(true);
```

Comment:
why you started a loading if you have condition and your request could not be executed. 
Move right before the request